src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -407,6 +418,10 @@ private void createMesosTask(Offer offer, Request request) {
     LOGGER.info("Launching task " + taskId.getValue() + " with URI " +
                 joinPaths(jenkinsMaster, SLAVE_JAR_URI_SUFFIX));
 
+    if (isExistingTask(offer, taskId)) {
```

Comment:
Here's one theory. On line `272`, you remove the request. However, there's no hashCode method and if anything changes inside it, then the default hashCode will change rendering it unremovable. Then when you loop through the requests again, it will get matched again and launched once more. I'm not too sure about that though, all I know is that this PR does fix the issue.

With this PR, you subvert the entire issue using the assumption that taskId is unique. On line 586, if you launch a task that is already in the list of `results`, then why should we launch it again? It will only become an orphan. On line `592`, we check the master list of tasks that are running or finished and should never run again. This prevents another class of orphaned mesos tasks.